core/templates/pages/topics-and-skills-dashboard-page/skills-list/skills-list.directive.spec.ts

```diff
@@ -0,0 +1,173 @@
+    var MockTopicsAndSkillsDashboardBackendApiService = {
+      mergeSkills: () => {
+        var deferred = $q.defer();
+        deferred.resolve();
+        return deferred.promise;
+      }
+    };
+    EditableTopicBackendApiService =
+        $injector.get('EditableTopicBackendApiService');
+    directive = $injector.get('skillsListDirective')[0];
+
+    ctrl = $injector.instantiate(directive.controller, {
+      $scope: $scope,
+      TopicsAndSkillsDashboardBackendApiService:
+      MockTopicsAndSkillsDashboardBackendApiService,
+      $uibModal
+    });
+  }));
+
+  it('should init the controller', function() {
+    ctrl.$onInit();
+    const skillHeadings = [
+      'index', 'description', 'worked_examples_count',
+      'misconception_count', 'status', 'options'];
+
+    expect(ctrl.SKILL_HEADINGS).toEqual(skillHeadings);
+  });
+
+  it('should return skill editor url', function() {
+    const skillId1 = 'uXcdsad3f42';
+    const skillId2 = 'aEdf44DGfre';
+    expect(ctrl.getSkillEditorUrl(skillId1)).toEqual(
+      '/skill_editor/uXcdsad3f42');
+    expect(ctrl.getSkillEditorUrl(skillId2)).toEqual(
+      '/skill_editor/aEdf44DGfre');
+  });
+
+  it('should open the delete skill modal', function() {
+    var modalSpy = spyOn($uibModal, 'open').and.callThrough();
+    ctrl.deleteSkill('dskfm4');
+    expect(modalSpy).toHaveBeenCalled();
+  });
+
+  it('should return serial number for skill', function() {
+    ctrl.getPageNumber = function() {
+      return 0;
+    };
+    ctrl.getItemsPerPage = function() {
+      return 10;
+    };
+    expect(ctrl.getSerialNumberForSkill(2)).toEqual(3);
+    ctrl.getPageNumber = function() {
+      return 3;
+    };
+    ctrl.getItemsPerPage = function() {
+      return 15;
+    };
+    expect(ctrl.getSerialNumberForSkill(2)).toEqual(48);
+  });
+
+  it('should reinitialize the page after successfully deleting the skill',
+    function() {
+      spyOn($uibModal, 'open').and.returnValue({
+        result: $q.resolve()
+      });
+      spyOn($rootScope, '$broadcast');
+
+      var skillId = 'CdjnJUE332dd';
+      var url = '/skill_editor_handler/data/' + skillId;
+      $httpBackend.expectDELETE(url).respond(200);
+      ctrl.deleteSkill(skillId);
+
+      $httpBackend.flush();
+      $timeout.flush();
+      expect($rootScope.$broadcast).toHaveBeenCalledWith(
+        'topicsAndSkillsDashboardReinitialized');
+    });
+
+  it('should reinitialize the page after merging the skill',
+    function() {
+      spyOn($uibModal, 'open').and.returnValue({
+        result: $q.resolve({
+          skill: {id: '1'},
+          supersedingSkillId: '2'
+        })
+      });
+
+      spyOn($rootScope, '$broadcast').and.callThrough();
+      $scope.getMergeableSkillSummaries = function() {
+        return [{id: 'dnfsdk', version: 1}];
+      };
+      $scope.getSkillsCategorizedByTopics = function() {};
+      var skillId = 'CdjnJUE332dd';
+
+      ctrl.mergeSkill(skillId);
+      $timeout.flush(100);
```

Comment:
Does this result in any flakiness in the frontend tests?

Reply:
No, I don't think so? They have been passing consistently.